src/file.js

```diff
@@ -503,27 +508,41 @@ File.prototype.createReadStream = function(options) {
       var start = is.number(options.start) ? options.start : '0';
       var end = is.number(options.end) ? options.end : '';
 
-      reqOpts.headers = {
-        Range: 'bytes=' + (tailRequest ? end : start + '-' + end),
-      };
+      reqOpts.headers.Range = `bytes=${tailRequest ? end : `${start}-${end}`}`;
     }
 
-    var requestStream = self.requestStream(reqOpts);
-    var validateStream;
+    var requestStream = self
+      .requestStream(reqOpts)
+      .on('error', err => {
+        throughStream.destroy(err);
+      })
+      .on('response', res => {
+        throughStream.emit('response', res);
+        common.util.handleResp(null, res, null, onResponse);
+      })
+      .resume();
+
+    throughStream.on('error', () => {
+      // An error can occur before the request stream has been created
+      // (during authentication)
+      if (requestStream.abort) {
+        requestStream.abort();
```

Comment:
I get lost around here, why do we need to call abort? It seems like if an error occurred it was either because of authentication (in which case there would be nothing to abort) or a request error, which the request failed so I'm not sure what's left to abort?

Reply:
Old code, don't remember for sure. I'll remove it.

Reply:
Done.